### PR TITLE
NumPy random generators

### PR DESCRIPTION
**Pull request by albi3ro**
Currently, the recommended way of creating random numbers with NumPy involves creating a local generator object, and then querying that object for random numbers

```python
import numpy as np
rng = np.default_rng()
rng.random((3,2))
```

Currently, we cannot use this method with PennyLane wrapped NumPy.  This PR adds a wrapped generator object and  `default_rng` generator constructor to PennyLane's NumPy random.  It also import the bitstring generators into the random namespace.  Now we can call

```python
from pennylane import numpy as np
rng = np.default_rng()
rng.random((3,2))
```

and get  an array with a `requires_grad` attribute,

### CODE REVIEW

**Comment by josh146** [comment]
@albi3ro, I'll attach the `WIP` label until tests are added 🙂

File: pennylane/numpy/random.py [hunk 1]
```diff
@@ -15,7 +15,44 @@
...
+    # pylint: disable=missing-class-docstring
+    class Generator(_random.Generator):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+            self.__doc__ = "PennyLane wrapped NumPy Generator object\n" + super().__doc__
+
+            for name in dir(_random.Generator):
+                if name[0] != "_":
+                    self.__dict__[name] = tensor_wrapper(getattr(super(), name))
```
Comment (josh146) [inline pennylane/numpy/random.py:40]: Nice! I tested this out, it works really well.

File: tests/numpy/test_numpy_random.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+        mat1 = rng1.random(size=size)
+        mat2 = rng2.random(size=size)
+
+        assert np.all(mat1 == mat2)
+
+        mat1_2 = rng1.normal(size=size)
+        mat2_2 = rng2.normal(size=size)
+
+        assert np.all(mat1_2 == mat2_2)
```
Comment (josh146) [inline tests/numpy/test_numpy_random.py:117]: Great tests!

File: pennylane/numpy/random.py (hunk 1 above)
Comment (josh146) [reply pennylane/numpy/random.py:40]: I had a look through all methods - it looks like most of them do return `ndarray` objects, so it should be good 👍 Are we inadvertently wrapping any that _don't_ return arrays?

**Review by josh146 — APPROVED** [review]
Nice work @albi3ro! This is a great add. Impressive testing as well!

My only big question mark is the `packaging` package, to make sure we don't cause import errors for end-users

File: pennylane/numpy/random.py [hunk 3]
```diff
@@ -15,7 +15,44 @@
 This package provides a wrapped version of autograd.numpy.random, such that
 it works with the PennyLane :class:`~.tensor` class.
 """
+from packaging.version import parse as parse_version
+
 from autograd.numpy import random as _random
-from .wrapper import wrap_arrays
+from numpy import __version__ as np_version
+from numpy.random import MT19937, PCG64, Philox, SFC64  # pylint: disable=unused-import
```
[the remark this answers is not in the document]
Comment (albi3ro) [reply pennylane/numpy/random.py:22]: We want these in the namespace, so someone can call `qml.numpy.random.MT19937`, instead of having to call `qml.numpy.random._random.MT19937`

File: pennylane/numpy/random.py (hunk 1 above)
Comment (albi3ro) [reply pennylane/numpy/random.py:40]: The methods will return a single scalar if not called with a size.  If a size is provided, then it returns an `ndarray`.  I've thought about explicitly writing out the names of all the generators that need to be wrapped, but decided against it.
